supervisor/addons/model.py

```diff
@@ -538,15 +538,22 @@ def need_build(self) -> bool:
         return ATTR_IMAGE not in self.data
 
     @property
-    def map_volumes(self) -> dict[str, bool]:
-        """Return a dict of {volume: read-only} from add-on."""
+    def map_volumes(self) -> dict[str, dict]:
```

Comment:
Create a dataclass to handle this:
```
FolderMapping:
  path: Path
  mode: RW/RO
  type: MappingType.XY
```

Reply:
Thanks for the feedback! I updated the pull request to use a FolderMapping dataclass